pkg/commands/workload.go

```diff
@@ -229,6 +230,8 @@ func (opts *WorkloadOptions) ApplyOptionsToWorkload(ctx context.Context, workloa
 		workload.Spec.MergeSourceImage(opts.SourceImage)
 	}
 
+	workload.Spec.MergeSubPath(opts.SubPath)
```

Comment:
Only update the sub path when a user explicitly set the flag. Without this check, all invocations that don't set the `--sub-path` flag will inadvertently unset the existing value because the default is `""`

```suggestion
	if cli.CommandFromContext(ctx).Flags().Changed(cli.StripDash(flags.SubPathFlagName)) {
		workload.Spec.MergeSubPath(opts.SubPath)
	}
```